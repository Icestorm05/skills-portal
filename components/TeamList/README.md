# Team List

## Functionality

The team list component displays a list of all the profiles passed in. The component will display the full name of the person who the profile belongs to, their job title, and will use the Employee ID as a route link parameter.

## Usage

### Props

`<team-list>` supports the following custom component attributes:

| attribute | type | description
| --- | --- | ---
| `lazy` | Boolean *optional* | Whether to render the subgroup lazily. Defaults to `true`.
| `team` | Array | The array of team members.
| `team-skills` | Array | The array of team members' skills.

### Schemas

The `team` prop requires the following object to exist within the array:

| attribute | type | description
| --- | --- | ---
| `EmployeeId` | String | The Employee ID of the profile.
| `FullName` | String | The full name of the person who the profile belongs to.
| `JobTitle` | String | The job title of the profile.

The `team-skills` prop requires the following object to exist within the array:

| attribute | type | description
| --- | --- | ---
| `diff` | Number | How many differences exist between skills and reviewed skills.
| `EmployeeId` | String | The Employee ID which maps to a profile.
| `skills` | Array *optional* | The array of skills. The top level should match the skill category schema, and **must** eventually lead down to a skill, saved skill or approved skill schema object.

#### Skill Category Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill category.
| `children` | Array | An array of further skill or skill categories. **Must** eventually lead down to a skill or saved skill schema object.

#### Skill Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill.
| `description` | String *optional* | The description of the skill.
| `SkillAttributeId` | Number | The unique attribute ID of the skill.

#### Saved Skill Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill.
| `description` | String *optional* | The description of the skill.
| `SkillAttributeId` | Number | The attribute ID of the skill.
| `EmployeeSkillAttributeId` | Number | A link to the SkillAttributeId.
| `EmployeeId` | Number | The ID of the employee who has the skill.
| `EmployeeSkillId` | Number | The unique ID of the saved skill.
| `SkillCurrencyId` | Number | The ID of the currency. 2 maps to `Current`. 3 maps to `Fading`. 4 maps to `Historic`.
| `SkillLevelId` | Number | The ID of the level. 2 maps to `Awareness`. 3 maps to `Practitioner`. 4 maps to `Expert`.

#### Approved Skill Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill.
| `description` | String *optional* | The description of the skill.
| `SkillAttributeId` | Number | The attribute ID of the skill.
| `EmployeeApproveSkillAttributeId` | Number | A link to the SkillAttributeId.
| `EmployeeApprovedSkillId` | Number | The unique ID of the approved skill.
| `EmployeeId` | Number | The ID of the employee who has the skill.
| `SkillCurrencyId` | Number | The ID of the currency. 2 maps to `Current`. 3 maps to `Fading`. 4 maps to `Historic`.
| `SkillLevelId` | Number | The ID of the level. 2 maps to `Awareness`. 3 maps to `Practitioner`. 4 maps to `Expert`.

### Example

```html
<team-list :team="team"/>
```

```ts
import Component from 'nuxt-class-component';
import Vue from 'vue';

import TeamList from '~/components/TeamList/TeamList.vue';

@Component({
    components: {
        TeamList,
    },
})
export default class GenericComponent extends Vue {
    private team = [{
        EmployeeId: 1,
        FullName: 'John Doe',
        JobTitle: 'Innovation Developer',
    }, {
        EmployeeId: 2,
        FullName: 'Jane Doe',
        JobTitle: 'Junior Innovation Developer',
    }]
}
```