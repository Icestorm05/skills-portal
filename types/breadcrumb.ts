import { Location } from 'vue-router';

export interface IBreadcrumb {
    link: Location;
    text: string;
}