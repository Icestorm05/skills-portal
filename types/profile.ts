export interface IProfile {
    Biography: string;
    Company: string;
    Director: string;
    EmailAddress: string;
    EmployeeId: number;
    FirstLineReporting: string;
    FullName: string;
    JobTitle: string;
    KnownAs: string;
    MainStaffNumber: string;
    SecondLineReporting: string;
    Surname: string;
    children: IProfile[];
}