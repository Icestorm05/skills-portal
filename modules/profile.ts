import { IProfile } from "types/profile";

/**
 * Calculates how deep a profile is in a hierarchy of profiles.
 * @param profiles {IProfile[]} The hierarchy of profiles.
 * @param profileToFind {IProfile} The profile to find.
 * @param level {number?} The depth level when no profile is found. Defaults to 0.
 * @return {number} The depth level.
 */
export const deepLevel = (
  profiles: IProfile[],
  profileToFind: IProfile,
  level: number = 0
): number => {
  return profiles.reduce((depth, profile: IProfile) => {
    if (profile.EmployeeId === profileToFind.EmployeeId) {
      return level + 1;
    } else if (profile.children) {
      const deepDepth = deepLevel(profile.children, profileToFind, level + 1);
      if (deepDepth) {
        return deepDepth;
      }
    }
    return depth;
  }, 0);
};

/**
 * Searches a hierarchy of profiles for one that matches a specified employee ID.
 * @param profiles {IProfile[]} The profile hierarchy.
 * @param id {number} The employee ID to find.
 * @return {IProfile | null} The matched profile, or null if not found.
 */
export const findProfile = (
  profiles: IProfile[],
  id: number
): IProfile | null => {
  return profiles.reduce((matchedProfile: IProfile | null, currProfile) => {
    if (!matchedProfile) {
      if (currProfile.EmployeeId === id) {
        return currProfile;
      } else if (currProfile.children) {
        return findProfile(currProfile.children, id);
      } else {
        return null;
      }
    }
    return matchedProfile;
  }, null);
};
