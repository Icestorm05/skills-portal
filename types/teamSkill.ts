import { ISkill } from 'types/skill';

export interface ITeamSkill {
    diff: number;
    EmployeeId: number;
    skills: ISkill[];
}
