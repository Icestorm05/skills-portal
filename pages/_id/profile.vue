<template lang="pug">
    #ProfilePage
        user-profile(:profile="profile"
                     :disabled="inProgress"
                     :editable="currentDepth === 1"
                     @updated-biography="updateBiography")
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Getter, State } from "vuex-class";

import { IProfile } from "types/profile";

import UserProfile from "~/components/UserProfile/UserProfile.vue";

@Component({
  components: {
    UserProfile
  }
})
export default class ProfilePage extends Vue {
  @Getter("profile/current") private profile: IProfile;
  @Getter("profile/currentDepth") private currentDepth: number;

  @Getter("progress/isVisible") private inProgress: boolean;

  private async updateBiography(Biography: string) {
    try {
      this.$store.commit("progress/show");
      await this.$store.dispatch("profile/edit", {
        ...this.profile,
        Biography
      });
      this.$store.commit("snackbar/show", {
        message: "Successfully updated bio.",
        color: "success"
      });
    } catch (err) {
      this.$store.commit("snackbar/show", {
        message: "Failed to update bio.",
        color: "error"
      });
    } finally {
      this.$store.commit("progress/hide");
    }
  }
}
</script>
