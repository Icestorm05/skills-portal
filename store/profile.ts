import { IProfile } from "types/profile";
import { deepLevel, findProfile } from "~/modules/profile";

const defaultProfile = {
  Biography: "",
  Director: "",
  EmailAddress: "",
  EmployeeId: 0,
  FirstLineReporting: "",
  FullName: "",
  JobTitle: "",
  KnownAs: "",
  SecondLineReporting: "",
  Surname: "",
  children: []
};

export const state = () => ({ ...defaultProfile });

export const mutations = {
  set(currentProfile: IProfile, profile: IProfile) {
    Object.keys(currentProfile).forEach(
      key => (currentProfile[key] = profile[key])
    );
  },
  updateBiography(currentProfile: IProfile, biography: string) {
    currentProfile.Biography = biography;
  },
  reset(currentProfile: IProfile) {
    Object.keys(currentProfile).forEach(
      key => (currentProfile[key] = defaultProfile[key])
    );
  }
};

export const actions = {
  async get({ commit }) {
    const data = await this.$axios.$get("/profile");
    commit("set", data.profile);
  },
  async edit({ commit }, profile: IProfile) {
    const url = `/employees/${profile.EmployeeId}`;
    await this.$axios.$put(url, { employee: profile });
    commit("updateBiography", profile.Biography);
  }
};

export const getters = {
  current(currentProfile: IProfile, localGetters, globalState) {
    const id = globalState.route
      ? parseInt(globalState.route.params.id, 10)
      : 0;
    const profile = findProfile(
      [currentProfile],
      id || currentProfile.EmployeeId
    );
    return profile ? profile : currentProfile;
  },
  currentDepth(currentProfile: IProfile, localGetters) {
    return localGetters.current.EmployeeId
      ? deepLevel([currentProfile], localGetters.current)
      : 0;
  },
  hasTeam(currentProfile: IProfile, localGetters) {
    return localGetters.current.children.length ? true : false;
  }
};
