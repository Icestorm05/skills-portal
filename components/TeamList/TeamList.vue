<template lang="pug">
    v-container(fluid
                grid-list-md
                v-if="team.length")#TeamList
        v-list(two-line).elevation-12
            v-list-group(v-for="teamMember in team"
                         :key="teamMember.EmployeeId"
                         :lazy="lazy"
                         :prepend-icon="icon(teamMember.EmployeeId)"
                         :append-icon="$vuetify.breakpoint.name === 'xs' ? '' : 'keyboard_arrow_down'"
                         no-action).TeamMember
                v-list-tile(slot="activator")
                    v-list-tile-content
                        v-list-tile-title.TeamMemberFullName {{ teamMember.FullName }}
                        v-list-tile-sub-title.TeamMemberJobTitle {{ teamMember.JobTitle }}
                    v-list-tile-avatar.TeamMemberAvatar.hidden-xs-only
                        avatar-image(:image-name="teamMember.MainStaffNumber" :renderAsBase64="true")
                v-list-tile(nuxt
                            :to="{ name: 'id-skills-unreviewed', params: { id: teamMember.EmployeeId } }").TeamMemberUnreviewedSkills
                    v-list-tile-content
                        v-list-tile-title View Unreviewed Skills
                    v-list-tile-action
                        v-badge(color="warning"
                                left).TeamMemberUnreviewedSkillBadge
                            span(slot="badge"
                                 v-if="diff(teamMember.EmployeeId)") {{ diff(teamMember.EmployeeId) }}
                            v-icon list
                v-list-tile(nuxt
                            :to="{ name: 'id-skills-reviewed', params: { id: teamMember.EmployeeId } }").TeamMemberReviewedSkills
                    v-list-tile-content
                        v-list-tile-title View Reviewed Skills
                    v-list-tile-action
                        v-icon playlist_add_check
                v-list-tile(nuxt
                            :to="{ name: 'id-profile', params: { id: teamMember.EmployeeId } }").TeamMemberProfile
                    v-list-tile-content
                        v-list-tile-title View Profile
                    v-list-tile-action
                        v-icon person
                v-list-tile(nuxt
                            :to="{ name: 'id-team', params: { id: teamMember.EmployeeId } }"
                            v-if="teamMember.children.length").TeamMemberTeam
                    v-list-tile-content
                        v-list-tile-title View Team
                    v-list-tile-action
                        v-icon people
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Prop } from "vue-property-decorator";

import { IProfile } from "types/profile";
import { ITeamSkill } from "types/teamSkill";

import VBadge from "vuetify/es5/components/VBadge";
import { VContainer } from "vuetify/es5/components/VGrid";
import VIcon from "vuetify/es5/components/VIcon";
import VList, {
  VListGroup,
  VListTile,
  VListTileAction,
  VListTileAvatar,
  VListTileContent,
  VListTileSubTitle,
  VListTileTitle
} from "vuetify/es5/components/VList";
import AvatarImage from "~/components/AvatarImage/AvatarImage.vue";

@Component({
  components: {
    AvatarImage,
    VBadge,
    VContainer,
    VIcon,
    VList,
    VListGroup,
    VListTile,
    VListTileAction,
    VListTileAvatar,
    VListTileContent,
    VListTileSubTitle,
    VListTileTitle
  }
})
export default class TeamList extends Vue {
  @Prop({ type: Boolean, default: true })
  private lazy: boolean;
  @Prop({ type: Array, required: true })
  private team: IProfile[];
  @Prop({ type: Array, required: true })
  private teamSkills: ITeamSkill[];

  private diff(EmployeeId: number) {
    const skills = this.skills(EmployeeId);
    return skills ? skills.diff : 0;
  }

  private icon(EmployeeId: number) {
    const skills = this.skills(EmployeeId);
    return skills && skills.diff ? "warning" : "verified_user";
  }

  private skills(EmployeeId: number) {
    return this.teamSkills.find(
      teamSkill => teamSkill.EmployeeId === EmployeeId
    );
  }
}
</script>
