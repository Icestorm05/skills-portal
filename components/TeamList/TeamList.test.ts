import TeamList from "./TeamList.vue";
import { mount } from "@vue/test-utils";

import { IProfile } from "types/profile";
import { ITeamSkill } from "types/teamSkill";

declare const expect: jest.Expect;

const defaultProfile: IProfile = {
  Biography: "",
  Company: "",
  Director: "",
  EmailAddress: "",
  EmployeeId: 0,
  FirstLineReporting: "",
  FullName: "",
  JobTitle: "",
  KnownAs: "",
  MainStaffNumber: "",
  SecondLineReporting: "",
  Surname: "",
  children: []
};

const profile: IProfile = {
  ...defaultProfile,
  EmployeeId: 1,
  FullName: "John Doe"
};

const profile2: IProfile = {
  ...defaultProfile,
  EmployeeId: 2,
  FullName: "Jane Doe"
};

const profile3: IProfile = {
  ...defaultProfile,
  EmployeeId: 3,
  FullName: "Jake Doe",
  children: [
    {
      ...defaultProfile,
      EmployeeId: 4,
      FullName: "Janet Doe"
    }
  ]
};

const team: IProfile[] = [profile, profile2, profile3];

const teamSkills1 = {
  diff: 0,
  EmployeeId: 1,
  skills: []
};
const teamSkills2 = {
  diff: 5,
  EmployeeId: 2,
  skills: []
};
const teamSkills3 = {
  diff: 1,
  EmployeeId: 3,
  skills: []
};

const teamSkills: ITeamSkill[] = [teamSkills1, teamSkills2, teamSkills3];

const stubs = {
  "nuxt-link": "<a></a>"
};

const mocks = {
  $vuetify: {
    breakpoint: "sm"
  }
};

describe("Team List", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { team, teamSkills }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("snapshot has not changed", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { team, teamSkills }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("team list does not display with blank team array", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { team: [], teamSkills }
    });
    expect(wrapper.find("#TeamList").exists()).toBeFalsy();
  });

  test("renders all team list member components", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { team, teamSkills }
    });
    expect(wrapper.findAll(".TeamMember").length).toEqual(3);
  });

  test("skills fn returns correct skills object", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { team, teamSkills }
    });
    expect((wrapper.vm as any).skills(1).diff).toEqual(teamSkills1.diff);
    expect((wrapper.vm as any).skills(2).diff).toEqual(teamSkills2.diff);
    expect((wrapper.vm as any).skills(3).diff).toEqual(teamSkills3.diff);
    expect((wrapper.vm as any).skills(4)).toBeUndefined();
  });

  test("displays team member names within list", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const names = wrapper.findAll(".TeamMember .TeamMemberFullName");
    expect(names.length).toEqual(3);
    expect(names.at(0).text()).toEqual(profile.FullName);
    expect(names.at(1).text()).toEqual(profile2.FullName);
    expect(names.at(2).text()).toEqual(profile3.FullName);
  });

  test("displays team member job titles within list", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const jobs = wrapper.findAll(".TeamMember .TeamMemberJobTitle");
    expect(jobs.length).toEqual(3);
    expect(jobs.at(0).text()).toEqual(profile.JobTitle);
    expect(jobs.at(1).text()).toEqual(profile2.JobTitle);
    expect(jobs.at(2).text()).toEqual(profile3.JobTitle);
  });

  test("displays team member avatars within list", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const avatars = wrapper.findAll(".TeamMember .TeamMemberAvatar img");
    expect(avatars.length).toEqual(3);
    expect(avatars.at(0).attributes().src).toEqual("/765-default-avatar.jpg");
    expect(avatars.at(1).attributes().src).toEqual("/765-default-avatar.jpg");
    expect(avatars.at(2).attributes().src).toEqual("/765-default-avatar.jpg");
  });

  test("displays correct icon if team member has unreviwed skills or not", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const icons = wrapper.findAll(
      ".TeamMember .list__group__header__prepend-icon .material-icons"
    );
    expect(icons.length).toEqual(3);
    expect(icons.at(0).text()).toEqual("verified_user");
    expect(icons.at(1).text()).toEqual("warning");
    expect(icons.at(2).text()).toEqual("warning");
  });

  test("does not display sublist by default", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    expect(wrapper.find(".TeamMemberUnreviewedSkills").isVisible()).toBeFalsy();
  });

  test("displays correct list of pages in sublist", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const teamMembers = wrapper.findAll(".TeamMember");
    expect(
      teamMembers
        .at(0)
        .find(".TeamMemberUnreviewedSkills")
        .exists()
    ).toBeTruthy();
    expect(
      teamMembers
        .at(0)
        .find(".TeamMemberReviewedSkills")
        .exists()
    ).toBeTruthy();
    expect(
      teamMembers
        .at(0)
        .find(".TeamMemberProfile")
        .exists()
    ).toBeTruthy();
  });

  test("does not display team page link if profile has no children", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const teamMembers = wrapper.findAll(".TeamMember");
    expect(
      teamMembers
        .at(0)
        .find(".TeamMemberTeam")
        .exists()
    ).toBeFalsy();
  });

  test("displays team page link if profile has children", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const teamMembers = wrapper.findAll(".TeamMember");
    expect(
      teamMembers
        .at(2)
        .find(".TeamMemberTeam")
        .exists()
    ).toBeTruthy();
  });

  test("displays correct number of skill differences as a badge on the unreviewed skills page link", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const teamMembers = wrapper.findAll(".TeamMember");
    expect(
      teamMembers
        .at(1)
        .find(".TeamMemberUnreviewedSkillBadge .badge__badge")
        .text()
    ).toEqual(teamSkills[1].diff.toString());
    expect(
      teamMembers
        .at(2)
        .find(".TeamMemberUnreviewedSkillBadge .badge__badge")
        .text()
    ).toEqual(teamSkills[2].diff.toString());
  });

  test("does not display icon as a badge if no skill differences on the unreviewed skills page link", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills }
    });
    const teamMembers = wrapper.findAll(".TeamMember");
    expect(
      teamMembers
        .at(0)
        .find(".TeamMemberUnreviewedSkillBadge .badge__badge")
        .exists()
    ).toBeFalsy();
  });

  test("defaults to displaying no badge when no matching team skills", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills: [] }
    });
    const teamMembers = wrapper.findAll(".TeamMember");
    expect(
      teamMembers
        .at(0)
        .find(".TeamMemberUnreviewedSkillBadge .badge__badge")
        .exists()
    ).toBeFalsy();
    expect(
      teamMembers
        .at(1)
        .find(".TeamMemberUnreviewedSkillBadge .badge__badge")
        .exists()
    ).toBeFalsy();
    expect(
      teamMembers
        .at(2)
        .find(".TeamMemberUnreviewedSkillBadge .badge__badge")
        .exists()
    ).toBeFalsy();
  });

  test("defaults to displaying verified user when no matching team skills", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team, teamSkills: [] }
    });
    const icons = wrapper.findAll(
      ".TeamMember .list__group__header__prepend-icon .material-icons"
    );
    expect(icons.length).toEqual(3);
    expect(icons.at(0).text()).toEqual("verified_user");
    expect(icons.at(1).text()).toEqual("verified_user");
    expect(icons.at(2).text()).toEqual("verified_user");
  });

  test("displays blank page with no team", () => {
    const wrapper = mount(TeamList, {
      mocks,
      stubs,
      propsData: { lazy: false, team: [], teamSkills: [] }
    });
    expect(wrapper.find("#TeamList").exists()).toBeFalsy();
  });
});
