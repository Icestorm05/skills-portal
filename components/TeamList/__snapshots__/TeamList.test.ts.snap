// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team List snapshot has not changed 1`] = `
<div class="container fluid grid-list-md" id="TeamList">
    <div class="list elevation-12 list--two-line">
        <div class="list__group TeamMember">
            <div class="list__group__header">
                <div class="list__group__header__prepend-icon">
                    <i aria-hidden="true" class="icon material-icons">verified_user</i>
                </div>
                <div>
                    <div class="list__tile">
                        <div class="list__tile__content">
                            <div class="list__tile__title TeamMemberFullName">John Doe</div>
                            <div class="list__tile__sub-title TeamMemberJobTitle"></div>
                        </div>
                        <div class="list__tile__avatar TeamMemberAvatar hidden-xs-only">
                            <div class="avatar" style="height: 40px; width: 40px;">
                                <img src="/765-default-avatar.jpg" alt="Avatar" class="AvatarImage">
                            </div>
                        </div>
                    </div>
                </div>
                <div class="list__group__header__append-icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="list__group__items list__group__items--no-action" style="display: none;"></div>
        </div>
        <div class="list__group TeamMember">
            <div class="list__group__header">
                <div class="list__group__header__prepend-icon">
                    <i aria-hidden="true" class="icon material-icons">warning</i>
                </div>
                <div>
                    <div class="list__tile">
                        <div class="list__tile__content">
                            <div class="list__tile__title TeamMemberFullName">Jane Doe</div>
                            <div class="list__tile__sub-title TeamMemberJobTitle"></div>
                        </div>
                        <div class="list__tile__avatar TeamMemberAvatar hidden-xs-only">
                            <div class="avatar" style="height: 40px; width: 40px;">
                                <img src="/765-default-avatar.jpg" alt="Avatar" class="AvatarImage">
                            </div>
                        </div>
                    </div>
                </div>
                <div class="list__group__header__append-icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="list__group__items list__group__items--no-action" style="display: none;"></div>
        </div>
        <div class="list__group TeamMember">
            <div class="list__group__header">
                <div class="list__group__header__prepend-icon">
                    <i aria-hidden="true" class="icon material-icons">warning</i>
                </div>
                <div>
                    <div class="list__tile">
                        <div class="list__tile__content">
                            <div class="list__tile__title TeamMemberFullName">Jake Doe</div>
                            <div class="list__tile__sub-title TeamMemberJobTitle"></div>
                        </div>
                        <div class="list__tile__avatar TeamMemberAvatar hidden-xs-only">
                            <div class="avatar" style="height: 40px; width: 40px;">
                                <img src="/765-default-avatar.jpg" alt="Avatar" class="AvatarImage">
                            </div>
                        </div>
                    </div>
                </div>
                <div class="list__group__header__append-icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="list__group__items list__group__items--no-action" style="display: none;"></div>
        </div>
    </div>
</div>
`;
